include/iota/api/service.hpp

```diff
@@ -74,13 +75,17 @@ class Service {
     json data;
     request.serialize(data);
 
-    auto url     = cpr::Url{ "http://" + host_ + ":" + std::to_string(port_) };
+    auto url     = cpr::Url{ host_ + ":" + std::to_string(port_) };
     auto body    = cpr::Body{ data.dump() };
     auto headers = cpr::Header{ { "Content-Type", "application/json" },
                                 { "Content-Length", std::to_string(body.size()) },
                                 { "X-IOTA-API-Version", APIVersion } };
-    auto res     = cpr::Post(url, body, headers, cpr::Timeout{ timeout_ * 1000 });
 
+    auto res     = cpr::Post(url, body, headers, cpr::Timeout{ timeout_ * 1000 });;
+    if(user_.compare("") != 0 && pass_.compare("") != 0 && host_.compare(0,5,"https") == 0){
+      auto auth = cpr::Authentication{user_, pass_};
+      res = cpr::Post(url, body, headers, cpr::Timeout{ timeout_ * 1000 }, auth);
```

Comment:
You are still posting twice in case of an authenticated request

Reply:
solved